src/uwtools/cli/run_forecast.py

```diff
@@ -22,9 +23,20 @@ def main() -> None:
     cli_helpers.setup_logging(
         log_file=args.log_file, log_name=name, quiet=args.quiet, verbose=args.verbose
     )
-    forecast_class = getattr(forecast, "%sForecast" % args.forecast_model)
-    experiment = forecast_class(args.config_file, args.machine, log_name=name)
-    experiment.run()
+    forecast_class = getattr(forecast, f"{args.forecast_model}Forecast")
+    try:
+        forecast_class.run(
+            config_file=args.config_file,
+            dry_run=args.dry_run,
+            forecast_app=args.forecast_app,
+            forecast_model=args.forecast_model,
+            log_file=args.log_file,
+            machine=args.machine,
+            quiet=args.quiet,
+            verbose=args.verbose,
+        )
+    except UWConfigError as e:
```

Comment:
I wonder if it might be time to build up a bit more of an exception hierarchy. For example, is the _only_ reason `run()` might fail a configuration error? If not, we should probably start raising (and handling) other exception types. For example, we could now (or soon) define `UWError(Exception)` as a base class for _all_ UW issues, redefine `UWConfigError(UWError)`, and catch `UWError` here, in case we are failing due to a UW issue other than a config issue. Is there an immediate use case for a new type of exception relating to _running_ and not just _configuring_?

Reply:
Ohh, that is a good point - we have taken over several other major features and we might as well finish taking over the exception handling entirely.

Reply:
I take it that a promising opportunity to diversify the UW exceptions did not present itself. Please feel free to resolve this comment if that's the case.

Reply:
I have an opportunity still to get it in, I'll see if I can

Reply:
Only if it looks genuinely useful!